### ISSUE TITLE
MigrationExecutor.getPendingMigrations() is not working properly

### ISSUE THREAD

**Issue by davilopez90** [typeorm/typeorm#5425]
**Issue type:**

[ ] question
[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[X] `cordova`
[X] `mongodb`
[X] `mssql`
[X] `mysql` / `mariadb`
[X] `oracle`
[X] `postgres`
[X] `cockroachdb`
[X] `sqlite`
[X] `sqljs`
[X] `react-native`
[X] `expo`

**TypeORM version:**

[X] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

I've detected a bug in [MigrationExecutor.getPendingMigrations()](https://github.com/typeorm/typeorm/blob/c0c8566d6fde6c3a794798d4dc9c27f20cbac762/src/migration/MigrationExecutor.ts#L93)
It's not filtering properly. In facto, it's not filtering at all, it's just returning allMigrations:
```
    /**
     * Returns an array of all pending migrations.
     */
    public async getPendingMigrations(): Promise<Migration[]> {
        const allMigrations = await this.getAllMigrations();
        const executedMigrations = await this.getExecutedMigrations();

        return allMigrations.filter(migration =>
            executedMigrations.find(
                executedMigration =>
                    executedMigration.name === migration.name
            )
        );
    }
```
This method should return actually pendingMigrations, I mean, the migrations contained in allMigrations but not in executedMigrations.
Solutions/proposals to solve this bug:

- Add a **!** before executedMigrations.find

```
    public async getPendingMigrations(): Promise<Migration[]> {
        const allMigrations = await this.getAllMigrations();
        const executedMigrations = await this.getExecutedMigrations();

        return allMigrations.filter(migration =>
            !executedMigrations.find(
                executedMigration =>
                    executedMigration.name === migration.name
            )
        );
    }
```

**Comment by captainjapeng** [comment]
In my case `getPendingMigrations` is returning all executed migrations similar to `getExecutedMigrations`.

Version: `"typeorm": "^0.2.24"`

**Comment by MartinodF** [comment]
I agree with @captainjapeng: instead of filtering out `executedMigrations`, thus returning only the pending ones, it's filtering only `executedMigrations`. Still happening in `0.2.25`, and the proposed fix (adding `!` before the filter call) is correct.

**Comment by bk3c** [comment]
Looks like this was fixed and merged in #6372; just waiting on a release.

**Comment by imnotjames** [comment]
The fix was released so I'm closing this out.
